qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1520A.py

```diff
@@ -598,8 +598,8 @@ def _get_sweep_steps_parser(response: str) -> Dict[str, Union[int, float]]:
         out_dict = cast(Dict[str, Union[int, float]], out_str)
```

Comment:
I would make outdict a typed_dict with actual types of the 5 different fields. 

Alternatively just avoiding copying out_str to out_dict would also make it cleaner and fix the mypy issue
```
out_dict: Dict[str, Union[int, float]] = {}
out_dict['chan'] = int(out_str['chan'])
...
out_dict['sweep_start'] = fixed_negative_float(out_str['sweep_end'])
...
```